frontend/src/pages/news/insideNewsPage.jsx

```diff
@@ -34,14 +37,156 @@ export default function InsideNewsPage() {
                 });
         } else {
             setArticle(newsData?.article);
-            setLikes(newsData?.likes);
         }
     }, []);
 
     useEffect(() => {
         if(article) setSections(Object.values(article?.article))
     }, [article]);
 
+    const [likeStatus, setLikeStatus] = useState(false);
+    const [initialLikeStatus, setInitialLikeStatus] = useState(false);
+    const elementRef = useRef(null);
+    const [hasRead, setHasRead] = useState(false);
+    const likeStatusRef = useRef(likeStatus);
+    const initialLikeStatusRef = useRef(initialLikeStatus);
+
+    const { user } = useContext(AuthContext)
+
+
+    useEffect(() => {
+        const observer = new IntersectionObserver(
+            ([entry]) => {
+                if (entry.isIntersecting && !hasRead) {
+                    saveInteraction('read');
```

Comment:
I don't like hardcoded strings like "read" here

Avoid that if possible.
For example you can create function `saveReadInteraction` and hardcode the string there (if that would be the only place it is used)